test/drivers/test_cqrrpt.cc

```diff
@@ -17,55 +17,83 @@ class TestCQRRPT : public ::testing::Test
 
     virtual void TearDown() {};
 
-    /// General test for CQRRPT:
-    /// Computes QR factorzation, and computes A[:, J] - QR.
-    template <typename T, typename RNG>
-    static void test_CQRRPT_general(int64_t m, int64_t n, int64_t k, int64_t d, int64_t nnz, T tol, std::tuple<int, T, bool> mat_type, RandBLAS::base::RNGState<RNG> state, uint64_t no_hqrrp) {
-
-        printf("|================================TEST CQRRPT GENERAL BEGIN===============================|\n");
-
-        int64_t size = m * n;
-        std::vector<T> A(size, 0.0);
-
+    template <typename T>
+    struct CQRRPTTestData {
+        std::vector<T> A;
         std::vector<T> R;
-        std::vector<int64_t> J(n, 0);
+        std::vector<int64_t> J;
+        std::vector<T> A_cpy;
 
-        RandLAPACK::util::gen_mat_type(m, n, A, k, state, mat_type);
+        CQRRPTTestData(int64_t m, int64_t n) :
+        A(m * n, 0.0), 
+        J(n, 0.0),  
+        A_cpy(m * n, 0.0) 
+        {}
+    };
 
-        std::vector<T> A_hat(size, 0.0);
-        std::copy(A.data(), A.data() + size, A_hat.data());
+    template <typename T, typename RNG>
+    static void computational_helper(int64_t m, int64_t n, CQRRPTTestData<T>& all_data) {
+        lapack::lacpy(MatrixType::General, m, n, all_data.A.data(), m, all_data.A_cpy.data(), m);
+    }
```

Comment:
Don't bother with this, my dude. In the place where you call this function, you can just do ``blas::copy(m*n, all_data.A.data(), 1, all_data.A_cpy.data(), 1)``; it's way more readable that way.

Reply:
Fixed.